File: app/src/main/java/id/husni/sultengcovid/adapter/DistrictAdapter.java
/*
 * Copyright (c) 2020.
 * Made with ❤ by Moh Husni Mubaraq
 * Not For Commercial Purpose
 */

package id.husni.sultengcovid.adapter;

import android.content.Context;
import android.content.Intent;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.Button;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import java.util.ArrayList;

import id.husni.sultengcovid.activity.DistrictDetailActivity;
import id.husni.sultengcovid.R;
import id.husni.sultengcovid.model.District;

public class DistrictAdapter extends RecyclerView.Adapter<DistrictAdapter.ViewHolder> {
    private final ArrayList<District> districts = new ArrayList<>();
    private final Context context;
    public DistrictAdapter(Context context) {
        this.context = context;
    }

    public void setDistricts(ArrayList<District> items) {
        if (districts.size() > 0) {
            districts.clear();
        }
        districts.addAll(items);
        notifyDataSetChanged();
    }

    @NonNull
    @Override
    public DistrictAdapter.ViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(context).inflate(R.layout.district_item_holder, parent, false);
        return new ViewHolder(view);
    }

    @Override
    public void onBindViewHolder(@NonNull DistrictAdapter.ViewHolder holder, int position) {
        holder.tvDistrictName.setText(districts.get(position).getDistrictName());
        holder.tvDistrictPositive.setText(String.valueOf(districts.get(position).getDistrictPositive()));
        holder.tvDistrictNegative.setText(String.valueOf(districts.get(position).getDistrictNegative()));
        holder.tvDistrictDeaths.setText(String.valueOf(districts.get(position).getDistrictDeaths()));
        holder.btnDetail.setOnClickListener(v -> {
            Intent intentDetail = new Intent(context, DistrictDetailActivity.class);
            intentDetail.putExtra(DistrictDetailActivity.EXTRA_PARSING_DATA,districts.get(position));
            context.startActivity(intentDetail);
        });
    }

    @Override
    public int getItemCount() {
        return districts.size();
    }

    public static class ViewHolder extends RecyclerView.ViewHolder {
        final TextView tvDistrictName;
        final TextView tvDistrictPositive;
        final TextView tvDistrictNegative;
        final TextView tvDistrictDeaths;
        final Button btnDetail;

        ViewHolder(@NonNull View itemView) {
            super(itemView);
            tvDistrictName = itemView.findViewById(R.id.tvDistrictName);
            tvDistrictPositive = itemView.findViewById(R.id.tvDistrictPositive);
            tvDistrictNegative = itemView.findViewById(R.id.tvDistrictNegative);
            tvDistrictDeaths = itemView.findViewById(R.id.tvDistrictDeaths);
            btnDetail = itemView.findViewById(R.id.btnDistrictDetail);
        }
    }
}

File: app/src/main/java/id/husni/sultengcovid/adapter/DistrictAdapter.kt
/*
 * Copyright (c) 2020.
 * Made with ❤ by Moh Husni Mubaraq
 * Not For Commercial Purpose
 */
package id.husni.sultengcovid.adapter

import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import com.afollestad.materialdialogs.MaterialDialog
import com.afollestad.materialdialogs.bottomsheets.BottomSheet
import com.afollestad.materialdialogs.bottomsheets.setPeekHeight
import com.afollestad.materialdialogs.customview.customView
import com.afollestad.materialdialogs.customview.getCustomView
import id.husni.sultengcovid.R
import id.husni.sultengcovid.model.District
import kotlinx.android.synthetic.main.district_item_holder.view.*

class DistrictAdapter(val context: Context?) :
    RecyclerView.Adapter<DistrictAdapter.ViewHolder>() {
    private val districtArray = ArrayList<District>()
    fun setData(items : ArrayList<District>){
        if (districtArray.size > 0){
            districtArray.clear()
        }
        districtArray.addAll(items)
        notifyDataSetChanged()
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
        val view : View = LayoutInflater.from(context).inflate(R.layout.district_item_holder, parent, false)
        return ViewHolder(view)
    }

    override fun getItemCount(): Int {
        return districtArray.size
    }

    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
        holder.bindData(districtArray[position])
    }

    class ViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        fun bindData(district: District) {
            with(itemView){
                tvDistrictName.text = district.districtName
                tvDistrictPositive.text = district.districtPositive.toString()
                tvDistrictNegative.text = district.districtNegative.toString()
                tvDistrictDeaths.text = district.districtDeaths.toString()
                btnDistrictDetail.setOnClickListener{
                    val dialog = MaterialDialog(context, BottomSheet()).show {
                        customView(R.layout.bottomsheet_district_detail, scrollable = true)
                        positiveButton(android.R.string.ok)
                        setPeekHeight(res = R.dimen.bottom_sheet_peek_height)
                        cornerRadius(res = R.dimen.bottom_sheet_corner_radius)
                    }
                    val view = dialog.getCustomView()
                    val districtName : TextView= view.findViewById(R.id.tvDistrictNameDetail)
                    val odp : TextView = view.findViewById(R.id.tvDistrictOdpDetail)
                    val pdp : TextView = view.findViewById(R.id.tvDistrictPdpDetail)
                    val positive : TextView = view.findViewById(R.id.tvDistrictPositiveDetail)
                    val negative : TextView = view.findViewById(R.id.tvDistrictNegativeDetail)
                    val death : TextView = view.findViewById(R.id.tvDistrictDeathsDetail)

                    districtName.text = district.districtName
                    odp.text = context.getString(R.string.odp, district.districtOdp.toString())
                    pdp.text = context.getString(R.string.pdp, district.districtPdp.toString())
                    positive.text = context.getString(R.string.positive_bs, district.districtPositive.toString())
                    negative.text = context.getString(R.string.negative_bs, district.districtNegative.toString())
                    death.text = context.getString(R.string.death_bs, district.districtDeaths.toString())
                }
            }
        }

    }

}